jsm.go

```diff
@@ -1008,17 +1013,34 @@ func (js *js) DeleteMsg(name string, seq uint64, opts ...JSOpt) error {
 		defer cancel()
 	}
 
-	if name == _EMPTY_ {
-		return ErrStreamNameRequired
+	return js.deleteMsg(o.ctx, name, msgDeleteRequest{Seq: seq, NoErase: true})
+}
+
+// SecureDeleteMsg deletes a message from a stream.
+// The deleted message is overwritten with random data
+func (js *js) SecureDeleteMsg(name string, seq uint64, opts ...JSOpt) error {
```

Comment:
If we go the option route, the implementation will simply have to set NoErase to opts.noErase (or whatever boolean you chose).